python/ray/data/tests/test_dataset_iterator.py

```diff
@@ -54,13 +54,12 @@ def test_basic_dataset_iter_rows(ray_start_regular_shared):
 
 
 def test_basic_dataset_pipeline(ray_start_regular_shared):
-    ds = ray.data.range(100).window(bytes_per_window=1).repeat()
+    ds = ray.data.range(100).window(bytes_per_window=1)
     it = ds.iterator()
-    for _ in range(2):
```

Comment:
By keeping using the `DatasetPipeline.iter_batches()`, the consumption is not repeatable.